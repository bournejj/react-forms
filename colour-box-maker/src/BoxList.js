import React, {useState} from "react";
import {v4 as uuid} from 'uuid'
import BoxItem from "./BoxItem";
import NewBoxItemForm from "./NewBoxItemForm";

const BoxList = () => {
    const INITIAL_STATE = [
        {id: "", colour: "", width: "", height: ""}
    ]
    const [items, setItems] = useState(INITIAL_STATE);
    
        const addItem = (colour, width, height) => {
            setItems(items => [...items, {id: uuid(), colour, width, height}])
    
        }

       const deleteBox = id => {
           setItems(items => {
               const copy = [...items];
               const item = id
               const filteredArray =  copy.filter(function(el) { return el.id !== item; });
               console.log(filteredArray)
               return filteredArray;

           })

       }
            






    return (
        <div>
            <h3> Box List</h3>
            <NewBoxItemForm addItem={addItem}/>
            <div>
                {items.map(({id, colour, width, height}) => <BoxItem deleteBox={deleteBox} id ={id} colour={colour} width={width} height={height}/>)}
                
            </div>
            




        </div>
    )
}

export default BoxList;

