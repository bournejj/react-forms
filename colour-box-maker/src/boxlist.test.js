import React from "react";
import {render, fireEvent} from "@testing-library/react";
import BoxList from "./BoxList.js";
import { query } from "express";

it("renders without crashing", function () {
    render(<BoxList/>);
})

it("matches snapshot", function () {
    const {asFragment} = render(<BoxList/>);
    expect(asFragment()).toMatchSnapshot();
    
})
it("it should add new item", function () {
    const {queryByText, getByLabelText} = render(<BoxList/>);
    const input = getByLabelText('colour');
    const btn = getByText('Add Box');
    expect(queryByText('colour: red')).not.toBeInTheDocument();
    fireEvent.change(input, {target: {value: 'red'}});
    fireEvent.click(btn);
    expect(queryByText('todo: red')).toBeInTheDocument();
    
    
})
