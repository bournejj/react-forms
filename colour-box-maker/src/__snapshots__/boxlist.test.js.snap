// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h3>
       Box List
    </h3>
    <form>
      <label
        for="colour"
      >
        colour
      </label>
      <input
        name="colour"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        width
      </label>
      <input
        name="width"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        height
      </label>
      <input
        name="height"
        type="text"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
    <div>
      <div
        class="Box"
      />
    </div>
  </div>
</DocumentFragment>
`;
